### ISSUE TITLE
Question about whisper server.

### ISSUE THREAD

**Issue by handsomelys** [PaddlePaddle/PaddleSpeech#4095]
## General Question

<img width="986" height="107" alt="Image" src="https://github.com/user-attachments/assets/ed21ce3d-41e3-436f-9efe-66782ca50fa9" />

As shown in the figure, the whisper-large service I deployed often outputs an exclamation mark "!" in the ASR. What may cause this situation?
The ckpt of whisper-large used comes from https://github.com/PaddlePaddle/PaddleSpeech/blob/develop/paddlespeech/resource/pretrained_models.py#L588
If whisper-tiny is used, this situation is almost not found. The tiny ckpt comes from
https://github.com/PaddlePaddle/PaddleSpeech/blob/develop/paddlespeech/resource/pretrained_models.py#L748

Looking forward to your reply!

中文翻译：

如图所示，我部署的whisper-large服务经常ASR出来的内容就是一个感叹号"!"，请问有什么可能会导致这种情况呢？
使用的whisper-large的ckpt来源于https://github.com/PaddlePaddle/PaddleSpeech/blob/develop/paddlespeech/resource/pretrained_models.py#L588
如果使用whisper-tiny倒几乎无发现这种情况，tiny ckpt来源于
https://github.com/PaddlePaddle/PaddleSpeech/blob/develop/paddlespeech/resource/pretrained_models.py#L748

期待您的回复！

**Comment by zxcd** [comment]
Firstly, we suggest that you call the whisper large model with command line, such as:
`paddlespeech whisper --task transcribe --input *.wav`
Then check the output, whether only output "!" . If still got same situation, you can update your wav to us to find out the problem.

**Comment by handsomelys** [comment]
> Firstly, we suggest that you call the whisper large model with command line, such as: `paddlespeech whisper --task transcribe --input *.wav` Then check the output, whether only output "!" . If still got same situation, you can update your wav to us to find out the problem.

Thanks for your reply. I have solved my problem. It was caused by the Paddle version and the dtype problem in whisper.py (various cuda errors appeared at the same time as the problem described in the issue).
Here I have a new problem. I found that the whisper large checkpoint here at https://github.com/PaddlePaddle/PaddleSpeech/blob/develop/paddlespeech/resource/pretrained_models.py#L748 is not the latest whisper-large-v3 from oai. Do you have the latest paddle version checkpoint of whisper-large-v3? If not, is there a tool that can convert torch model parameters to paddle parameters?

**Comment by zxcd** [comment]
`whisper-large-v3` is currently not supported. If you want to use v3 model, not only need to change model parameters to paddle parameters, but also need to change tokenizer part.

For model convert tool, the key point is to transpose the two-dimensional *.weight params. However, due to differences in parameters and structures, there is currently no common tool available and manual modifications are required.

For example:
```
import torch
import paddle

torch_model_dict = torch.load('weights.pth')["state_dict"]

paddle_model_state_dict={}
paddle_model_state_dict['state_dict']={}
for key, val in torch_model_dict.items():
    if key.endswith('weight') and val.ndim == 2:
        val = val.T
    # add some others transpose if need
    paddle_model_state_dict['state_dict'][key] = paddle.to_tensor(val.cpu().numpy())

# add other params in case if need, such as:
# paddle_model_state_dict['metadata'] = torch.load('weights.pth')["metadata"]

paddle.save(paddle_model_state_dict, 'weights.params')
```

**Comment by handsomelys** [comment]
Thank you for your reply! I have now manually converted a whisper-large-v3 ckpt, which can be correctly loaded into the current whisper network in paddlespeech. But the inference result of the model does not meet expectations. I have read the relevant issues of openai. The main difference between whisper-large-v3 and other versions is the number of mel-filters and the addition of a special token of yue in the tokenizer. I have now taken the following measures:
1. Torch params are converted to paddle params (can be loaded into the model, but I don’t know how to determine whether the conversion is correct);
2. Replaced mel_filter.npz so that n_mels=128 can be used;
3. Added a special token of |yue| in tokenizer.py, but my gpt2tokenizer was not replaced, and the one released by openai was tiktoken.
I would like to ask if there are any steps missing in my above approach. If it is convenient, I would like to contact you on infoflow or other means.

**Comment by handsomelys** [comment]
> `whisper-large-v3` is currently not supported. If you want to use v3 model, not only need to change model parameters to paddle parameters, but also need to change tokenizer part.
> 
> For model convert tool, the key point is to transpose the two-dimensional *.weight params. However, due to differences in parameters and structures, there is currently no common tool available and manual modifications are required.
> 
> For example:
> 
> ```
> import torch
> import paddle
> 
> torch_model_dict = torch.load('weights.pth')["state_dict"]
> 
> paddle_model_state_dict={}
> paddle_model_state_dict['state_dict']={}
> for key, val in torch_model_dict.items():
>     if key.endswith('weight') and val.ndim == 2:
>         val = val.T
>     # add some others transpose if need
>     paddle_model_state_dict['state_dict'][key] = paddle.to_tensor(val.cpu().numpy())
> 
> # add other params in case if need, such as:
> # paddle_model_state_dict['metadata'] = torch.load('weights.pth')["metadata"]
> 
> paddle.save(paddle_model_state_dict, 'weights.params')
> ```


Thank you for your reply! I have now manually converted a whisper-large-v3 ckpt, which can be correctly loaded into the current whisper network in paddlespeech. But the inference result of the model does not meet expectations. I have read the relevant issues of openai. The main difference between whisper-large-v3 and other versions is the number of mel-filters and the addition of a special token of yue in the tokenizer. I have now taken the following measures:
1. Torch params are converted to paddle params (can be loaded into the model, but I don’t know how to determine whether the conversion is correct);
2. Replaced mel_filter.npz so that n_mels=128 can be used;
3. Added a special token of |yue| in tokenizer.py, but my gpt2tokenizer was not replaced, and the one released by openai was tiktoken.
I would like to ask if there are any steps missing in my above approach. If it is convenient, I would like to contact you on infoflow or other means.

**Comment by zxcd** [comment]
Since we still not supported v3, it's still uncertain whether there are any issues at the moment. But if you need to check model correctness, it can be judged by running inference on a test set of an open-source dataset.  WER will be the best validation indicators. We also welcome you to contribute PR to us.

For contacting us, you can add us on our operated WeChat account, and she will add you to PaddleSpeech's WeChat communication group.

<img width="506" height="500" alt="Image" src="https://github.com/user-attachments/assets/627d446d-250c-4e45-852c-48b9276b46b5" />
